src/vcpkg/commands.z-applocal.cpp

```diff
@@ -585,17 +597,70 @@ namespace vcpkg
         }
 
         const auto target_installed_bin_dir =
-            real_filesystem.almost_canonical(target_installed_bin_setting->second, VCPKG_LINE_INFO);
-        const auto target_binary_path = real_filesystem.almost_canonical(target_binary->second, VCPKG_LINE_INFO);
+            fs.almost_canonical(target_installed_bin_setting->second, VCPKG_LINE_INFO);
         const auto decoded = decode_from_canonical_bin_dir(target_installed_bin_dir);
-        AppLocalInvocation invocation(real_filesystem,
+
+        // the first binary is special in that it might not be a DLL or might not exist
+        const Path target_binary_path = target_binary->second;
+        msg::print(LocalizedString::from_raw(target_binary_path)
+                       .append_raw(": ")
+                       .append_raw(MessagePrefix)
+                       .append(msgApplocalProcessing)
+                       .append_raw('\n'));
+
+        std::error_code ec;
+        auto dll_file = fs.open_for_read(target_binary_path, ec);
+        if (ec)
+        {
+            auto io_error = ec.message();
+            if (ec == std::errc::no_such_file_or_directory)
+            {
+                msg::print(Color::warning,
+                           LocalizedString::from_raw(target_binary_path)
+                               .append_raw(": ")
+                               .append_raw(WarningPrefix)
+                               .append_raw(io_error)
+                               .append_raw('\n'));
+            }
+            else
+            {
+                msg::print(Color::error,
+                           LocalizedString::from_raw(target_binary_path)
+                               .append_raw(": ")
+                               .append_raw(ErrorPrefix)
+                               .append_raw(io_error)
+                               .append_raw('\n'));
+            }
+
+            Checks::exit_fail(VCPKG_LINE_INFO);
+        }
+
+        auto maybe_dll_metadata = vcpkg::try_read_dll_metadata(dll_file).value_or_exit(VCPKG_LINE_INFO);
```

Comment:
Why not use `try_read_dll_metadata_required`?

Reply:
Because the whole point of this is to use the not required one.